### ISSUE TITLE
About the back-propagation of kd_loss

### ISSUE THREAD

**Issue by RuoyiDu** [huawei-noah/Data-Efficient-Model-Compression#18]
The kd_loss is used for the training of student network. However, it is also back propagated to the encoder during training, which is not same as the loss function in the paper.

**Comment by HantingChen** [comment]
https://github.com/huawei-noah/Data-Efficient-Model-Compression/blob/5e8f4e4cd361778a1d43772f4bb032de2d874393/DAFL/DAFL-train.py#L181

As we detach the gradient for the generated image, I think the kd_loss is not back propagated to the encoder during training.

**Comment by RuoyiDu** [comment]
> https://github.com/huawei-noah/Data-Efficient-Model-Compression/blob/5e8f4e4cd361778a1d43772f4bb032de2d874393/DAFL/DAFL-train.py#L181
> 
> As we detach the gradient for the generated image, I think the kd_loss is not back propagated to the encoder during training.

Thanks for your reply! I missed this part!
